docker/jenkins/build.sh

```diff
@@ -17,16 +17,22 @@ mkdir build && cd build
 # configure trilinos with dtk
 if [ -z "${CUDA_VERSION}" ]
 then
-    if [ "${SANITIZE}" == "undefined" ]
+    if [ "${COMPILER}" == "clang" ]
     then
-        source ../scripts/docker_clang_env.sh undefined_sanitizer
-        ../scripts/docker_cmake -D Trilinos_ENABLE_Fortran=OFF
-    elif [ "${SANITIZE}" == "thread" ]
-    then
-        source ../scripts/docker_clang_env.sh thread_sanitizer
-        ../scripts/docker_cmake -D Trilinos_ENABLE_Fortran=OFF
+        if [ "${SANITIZE}" == "undefined" ]
+        then
+            source ../scripts/docker_clang_env.sh undefined_sanitizer
+            ../scripts/docker_cmake -D Trilinos_ENABLE_Fortran=OFF
+        elif [ "${SANITIZE}" == "thread" ]
+        then
+            source ../scripts/docker_clang_env.sh thread_sanitizer
+            ../scripts/docker_cmake -D Trilinos_ENABLE_Fortran=OFF
+        else
+            ../scripts/docker_cmake -D Trilinos_ENABLE_COVERAGE_TESTING=ON
+        fi
     else
-        ../scripts/docker_cmake -D Trilinos_ENABLE_COVERAGE_TESTING=ON
+        source ../scripts/docker_gcc71_env.sh
+        ../scripts/docker_cmake -D Trilinos_CXX11_FLAGS="-std=c++17" -D Trilinos_ENABLE_OpenMP=OFF -D Tpetra_INST_OPENMP=OFF -D Kokkos_ENABLE_OpenMP=OFF
```

Comment:
If you set `Tpetra_INST_OPENMP=ON` but Kokkos's OpenMP is off, it will throw an error in configure. I see your point, we don't want any throws in the configuration, so may as well disable all.